### PR TITLE
endlines: init at 1.9.2

### PR DESCRIPTION
**Pull request by zedseven**
## Description of changes

`endlines` is a command-line utility for conversion of line-endings in files and entire directories.

The repository comes with a `tests` directory, but they're broken.

**Homepage:** https://github.com/mdolidon/endlines

This is my first contribution here, and I'm still learning Nix. Any feedback is welcome.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/en/endlines/package.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+in
+  stdenv.mkDerivation (finalAttrs: {
+    pname = "endlines";
+    version = version;
+
+    src = fetchFromGitHub {
+      owner = "mdolidon";
+      repo = "endlines";
+      rev = version;
+      sha256 = "sha256-M0IyY/WXR8qv9/qx5G0pG3EKqMoZAP3fJTZ6sSSMMyQ=";
```
Comment (eclairevoyant) [inline pkgs/by-name/en/endlines/package.nix:16]:
```suggestion
      hash = "sha256-M0IyY/WXR8qv9/qx5G0pG3EKqMoZAP3fJTZ6sSSMMyQ=";
```

File: pkgs/by-name/en/endlines/package.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+}: let
+  version = "1.9.2";
+in
+  stdenv.mkDerivation (finalAttrs: {
+    pname = "endlines";
+    version = version;
```
Comment (eclairevoyant) [inline pkgs/by-name/en/endlines/package.nix:10]:
```suggestion
    version = "1.9.2";
```

File: pkgs/by-name/en/endlines/package.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    src = fetchFromGitHub {
+      owner = "mdolidon";
+      repo = "endlines";
+      rev = version;
+      sha256 = "sha256-M0IyY/WXR8qv9/qx5G0pG3EKqMoZAP3fJTZ6sSSMMyQ=";
+    };
+
+    preInstall = ''
+      sed -i "s@/usr/local@$out@g" Makefile
```
Comment (eclairevoyant) [inline pkgs/by-name/en/endlines/package.nix:20]: this should go in a patch phase

File: pkgs/by-name/en/endlines/package.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+}: let
```
Comment (eclairevoyant) [inline pkgs/by-name/en/endlines/package.nix:5]: if you're using `finalAttrs`, remove the `let` block.

File: pkgs/by-name/en/endlines/package.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  version = "1.9.2";
+in
+  stdenv.mkDerivation (finalAttrs: {
+    pname = "endlines";
+    version = version;
+
+    src = fetchFromGitHub {
+      owner = "mdolidon";
+      repo = "endlines";
+      rev = version;
```
Comment (eclairevoyant) [inline pkgs/by-name/en/endlines/package.nix:15]:
```suggestion
      rev = finalAttrs.version;
```

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
Thanks for the initial contribution! Some feedback below

**Review by eclairevoyant — APPROVED** [review]

File: pkgs/by-name/en/endlines/package.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      mkdir -p $out/bin
+    '';
+
+    meta = with lib; {
+      homepage = "https://github.com/mdolidon/endlines";
+      description = "Easy conversion between new-line conventions";
+      license = licenses.asl20;
+      maintainers = with maintainers; [zedseven];
+      mainProgram = "endlines";
+      platforms = platforms.unix;
```
Comment (AndersonTorres) [inline pkgs/by-name/en/endlines/package.nix:31]:
```suggestion
    meta = {
      homepage = "https://github.com/mdolidon/endlines";
      description = "Easy conversion between new-line conventions";
      license = lib.licenses.asl20;
      maintainers = with lib.maintainers; [ zedseven ];
      mainProgram = "endlines";
      platforms = lib.platforms.unix;
```

https://nix.dev/guides/best-practices.html#with-scopes

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/en/endlines/package.nix [hunk 7]
```diff
@@ -0,0 +1,33 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+}:
+  stdenv.mkDerivation (finalAttrs: {
```
Comment (AndersonTorres) [inline pkgs/by-name/en/endlines/package.nix:6]: Fixup indentation.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Comment by eclairevoyant** [comment]
@ofborg eval

**Review by AndersonTorres — APPROVED** [review]

**Comment by AndersonTorres** [comment]
I will send this to the Discourse.

File: pkgs/by-name/en/endlines/package.nix [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+  src = fetchFromGitHub {
+    owner = "mdolidon";
+    repo = "endlines";
+    rev = finalAttrs.version;
+    hash = "sha256-M0IyY/WXR8qv9/qx5G0pG3EKqMoZAP3fJTZ6sSSMMyQ=";
+  };
+
+  postPatch = ''
+    sed -i "s@/usr/local@$out@g" Makefile
```
Comment (pbsds) [inline pkgs/by-name/en/endlines/package.nix:18]:
```suggestion
    substituteInPlace Makefile \
      --replace "/usr/local" "$out"
```

File: pkgs/by-name/en/endlines/package.nix
Comment (philiptaron) [inline pkgs/by-name/en/endlines/package.nix:1]: Run `nixpkgs-fmt` on this file. 🦺

**Review by philiptaron — APPROVED** [review]

File: pkgs/by-name/en/endlines/package.nix
Comment (philiptaron) [reply pkgs/by-name/en/endlines/package.nix:1]: Today I learned.

**Review by pbsds — APPROVED** [review]
Result of `nixpkgs-review pr 271173` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>endlines</li>
  </ul>
</details>

LGTM, thanks!

**Pull request merged**